scripts/yg-privacyenhanced.py

```diff
@@ -2,30 +2,64 @@
 from future.utils import iteritems
 
 import random
+import os
 
 from jmbase import get_log, jmprint
+from jmbase.support import lookup_appdata_folder
 from jmclient import YieldGeneratorBasic, ygmain, jm_single
-
+from configparser import ConfigParser
 
 # This is a maker for the purposes of generating a yield from held bitcoins
 # while maximising the difficulty of spying on blockchain activity.
 # This is primarily attempted by randomizing all aspects of orders
 # after transactions wherever possible.
 
-"""THESE SETTINGS CAN SIMPLY BE EDITED BY HAND IN THIS FILE:
-"""
+defaultygconfig = \
+    """
+[YIELDGENERATOR]
+# [string, 'reloffer' or 'absoffer'], which fee type to actually use
+ordertype = reloffer
+
+# [satoshis, any integer] / absolute offer fee you wish to receive for coinjoins (cj)
+cjfee_a = 500
+
+# [fraction, any str between 0-1] / relative offer fee you wish to receive based on a cj's amount
+cjfee_r = 0.00002
+
+# [fraction, 0-1] / variance around the average fee. Ex: 200 fee, 0.2 var = fee is btw 160-240
+cjfee_factor = 0.1
+
+# [satoshis, any integer] / the average transaction fee you're adding to coinjoin transactions
+txfee = 100
+
+# [fraction, 0-1] / variance around the average fee. Ex: 1000 fee, 0.2 var = fee is btw 800-1200
+txfee_factor = 0.3
+
+# [satoshis, any integer] / minimum size of your cj offer. Lower cj amounts will be disregarded
+minsize = 100000
+
+# [fraction, 0-1] / variance around all offer sizes. Ex: 500k minsize, 0.1 var = 450k-550k
+size_factor = 0.1
 
-ordertype = 'reloffer'  # [string, 'reloffer' or 'absoffer'], which fee type to actually use
-cjfee_a = 500             # [satoshis, any integer] / absolute offer fee you wish to receive for coinjoins (cj)
-cjfee_r = '0.00002'       # [fraction, any str between 0-1] / relative offer fee you wish to receive based on a cj's amount
-cjfee_factor = 0.1        # [fraction, 0-1] / variance around the average fee. Ex: 200 fee, 0.2 var = fee is btw 160-240
-txfee = 100               # [satoshis, any integer] / the average transaction fee you're adding to coinjoin transactions
-txfee_factor = 0.3        # [fraction, 0-1] / variance around the average fee. Ex: 1000 fee, 0.2 var = fee is btw 800-1200
-minsize = 100000          # [satoshis, any integer] / minimum size of your cj offer. Lower cj amounts will be disregarded
-size_factor = 0.1         # [fraction, 0-1] / variance around all offer sizes. Ex: 500k minsize, 0.1 var = 450k-550k
 gaplimit = 6
+"""
 
-# end of settings customization
+config = ConfigParser()
+config.read(os.path.join(lookup_appdata_folder(jm_single().APPNAME),"joinmarket.cfg"))
```

Comment:
If `jm_single().config` (`jm_single` is imported from `jmclient` in `yieldgenerator.py`, which is where the start up function `ygmain()` is called; the config is set after `load_program_config` has been called) is accessible, this processing is not needed:
Instead, you would just add the defaults to the defaultconfig string here:

https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/14f0f46b0edd441269aa80c9d9b703bf6f62b1a6/jmclient/jmclient/configure.py#L94-L352

 because the config is read in a way that automatically uses the defaults, if there is no setting in the file `<datadir>/joinmarket.cfg`.